src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AddMissingInterpolationToken.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    public abstract class AbstractAddMissingInterpolationTokenAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA2253";
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(RuleId,
+            new LocalizableResourceString(nameof(MicrosoftCodeQualityAnalyzersResources.AddMissingInterpolationTokenTitle), MicrosoftCodeQualityAnalyzersResources.ResourceManager, typeof(MicrosoftCodeQualityAnalyzersResources)),
+#pragma warning disable RS1032 // Define diagnostic message correctly - the analyzer wants a period after the existing question mark.
```

Comment:
That's funny, I recall having handle various punctuation in the initial implementation. I would recommend to create a ticket to fix the implementation and link it in the code, WDYT?

Reply:
→ #5310